src/Lykke.Service.PayAuth/Controllers/RegisterController.cs

```diff
@@ -10,14 +13,35 @@ namespace Lykke.Service.PayAuth.Controllers
     [Route("api/[controller]")]
     public class RegisterController : Controller
     {
+        private readonly IPayAuthService _payAuthService;
+        public RegisterController(IPayAuthService payAuthService)
+        {
+            _payAuthService = payAuthService;
+        }
         [HttpPost]
         [SwaggerOperation("Register")]
-        [ProducesResponseType(typeof(ErrorResponse), (int) HttpStatusCode.BadRequest)]
-        [ProducesResponseType(typeof(ErrorResponse), (int) HttpStatusCode.InternalServerError)]
-        [ProducesResponseType(typeof(void), (int) HttpStatusCode.OK)]
-        public Task<IActionResult> Register(string systemId, string clientId, string apiKey, string certificate)
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> Register(string systemId, string clientId, string apiKey, string certificate)
         {
-            throw new NotImplementedException();
+            try
+            {
+                var model = new NewPayAuthModel
+                {
+                    ClientId = clientId,
+                    SystemId = systemId,
+                    ApiKey = apiKey,
+                    Certificate = certificate
+                };
+                var sign = Mapper.Map<Core.Domain.PayAuth>(model);
+                await _payAuthService.AddAsync(sign);
+            }
+            catch (Exception exception)
+            {
+                return BadRequest(ErrorResponse.Create(exception.Message));
```

Comment:
I think it is more appropriate here to return InternalServerError rather than BadRequest. Also I'd not return exception.Message as an API error description.